lib/sinon/util/core/deep-equal.js

```diff
@@ -1,99 +1,46 @@
 "use strict";
 
-var div = typeof document !== "undefined" && document.createElement("div");
+var samsam = require("@sinonjs/samsam");
 
 function isReallyNaN(val) {
     return val !== val;
 }
 
-function isDOMNode(obj) {
-    var success = false;
-
-    try {
-        obj.appendChild(div);
-        success = div.parentNode === obj;
-    } catch (e) {
-        return false;
-    } finally {
-        try {
-            obj.removeChild(div);
-        } catch (e) {
-            // Remove failed, not much we can do about that
-        }
+var deepEqual = module.exports = function deepEqual(a, b, matcher) {
+    if (a === null && b === null) {
+        return true;
     }
 
-    return success;
-}
-
-function isElement(obj) {
-    return div && obj && obj.nodeType === 1 && isDOMNode(obj);
-}
-
-var deepEqual = module.exports = function deepEqual(a, b) {
     if (typeof a !== "object" || typeof b !== "object") {
         return isReallyNaN(a) && isReallyNaN(b) || a === b;
     }
 
-    if (isElement(a) || isElement(b)) {
-        return a === b;
-    }
-
-    if (a === b) {
-        return true;
-    }
-
-    if ((a === null && b !== null) || (a !== null && b === null)) {
-        return false;
-    }
-
-    if (a instanceof RegExp && b instanceof RegExp) {
-        return (a.source === b.source) && (a.global === b.global) &&
-            (a.ignoreCase === b.ignoreCase) && (a.multiline === b.multiline);
-    }
-
     if (a instanceof Error && b instanceof Error) {
         return a === b;
     }
 
-    var aString = Object.prototype.toString.call(a);
-    if (aString !== Object.prototype.toString.call(b)) {
+    if (Object.prototype.toString.call(a) !== Object.prototype.toString.call(b)) {
         return false;
     }
 
-    if (aString === "[object Date]") {
-        return a.valueOf() === b.valueOf();
-    }
-
-    var prop;
-    var aLength = 0;
-    var bLength = 0;
-
-    if (aString === "[object Array]" && a.length !== b.length) {
+    var haveDifferentKeys = Object.keys(a).sort().join() !== Object.keys(b).sort().join();
```

Comment:
I think this is straight-forward enough to put inline in the `if` statement without the single-use variable in my opinion.